util/util.js

```diff
@@ -119,15 +119,26 @@ function exitRoom(lobby, roomId, playerId) {
     var room = lobby.getRoomById(roomId)
     room.removePlayer(playerId)
     if (room.isEmpty()) {
+        console.log('Deleting room, since there is no players left in the room')
         lobby.removeRoom(roomId)
-        return undefined
+    }
+    else {
+        // add room to availableRooms list
+        // remove the room from fullRooms list
+        if (room.roomMembers.length < room.maxPlayersPerRoom) {
+            lobby.availableRooms.push(room)
+            lobby.fullRooms.forEach((fullRoom, index, object) => {
+                if (fullRoom.roomId === room.roomId) {
+                    object.splice(index, 1)
+                }
+            })
```

Comment:
Similar comment as before. I think availableRooms and fullRooms should be a set or a map instead of an array. This gets rid of the need to iterate through the entire list of rooms, just to delete one

Reply:
Yes, If long lived, just make it a map. It's far less error prone.